box-share-sdk/src/main/java/com/box/androidsdk/share/usx/fragments/CollaborationsFragment.java

```diff
@@ -0,0 +1,322 @@
+        } else {
+            if (mSelectRoleShareVM.getSelectedRole().getValue() != null && mSelectRoleShareVM.getCollaboration() != null) {
+                if (mSelectRoleShareVM.getSelectedRole().getValue() != mSelectRoleShareVM.getCollaboration().getRole()) { //this means user selected a different role.
+                    if (mSelectRoleShareVM.getSelectedRole().getValue() == BoxCollaboration.Role.OWNER) {
+                        AlertDialog dialog = new AlertDialog.Builder(getActivity()).setTitle(R.string.box_sharesdk_change_owner_alert_title)
+                                .setMessage(R.string.box_sharesdk_change_owner_alert_message)
+                                .setPositiveButton(android.R.string.yes, (d, which) -> {
+                                    showSpinner(R.string.box_sharesdk_fetching_collaborators, R.string.boxsdk_Please_wait);
+                                    mCollaborationsShareVM.updateOwner(mSelectRoleShareVM.getCollaboration());
+                                }).setNegativeButton(android.R.string.no, (d, which) -> {}).setIcon(android.R.drawable.ic_dialog_alert).create();
+                        dialog.show();
+                    } else {
+                        showSpinner();
+                        mCollaborationsShareVM.updateCollaboration(mSelectRoleShareVM.getCollaboration(), mSelectRoleShareVM.getSelectedRole().getValue());
+
+                    }
+                    mSelectRoleShareVM.setSelectedRole(null); //reset selected role
+                }
+            }
+        }
+
+        if (mCollaborationsShareVM.getCollaborations().getValue() == null) {
+            mCollaborationsShareVM.fetchItemInfo(mCollaborationsShareVM.getShareItem());
+        }
+
+
+    }
+
+    public void setCallback(CollaborationsFragmentCallback callback) {
+        this.mCallback = callback;
+    }
+
+    @Override
+    public void addResult(Intent data) {
+//        data.putExtra(CollaborationUtils.EXTRA_COLLABORATIONS, mCollaborations);
+        data.putExtra(CollaborationUtils.EXTRA_OWNER_UPDATED, mCollaborationsShareVM.isOwnerUpdated());
+        //I commented these out since this could be passing too much data inside an Intent.
+        super.addResult(data);
+    }
+
+    @Override
+    protected void setTitles() {
+        ActionbarTitleVM actionbarTitleVM = ViewModelProviders.of(getActivity()).get(ActionbarTitleVM.class);
+        actionbarTitleVM.setTitle(getString(R.string.box_sharesdk_shared_with));
+        actionbarTitleVM.setSubtitle(null);
+    }
+
+    @Override
+    public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+        BoxCollaboration collaboration = (BoxCollaboration) mCollaboratorsAdapter.getItem(position);
+        if (collaboration != null) {
+            ArrayList<BoxCollaboration.Role> rolesArr = getRoles();
+            String collabId = collaboration.getAccessibleBy().getId();
```

Comment:
null check

Reply:
agreed for this one. the getAccessibleBy() can return null.